### DISCUSSION TITLE
Add Spanish language to the UI

### DISCUSSION THREAD

**Discussion by laramaceo** [tnodir/fort#280] [General]
Hello.

I have discovered this great program recently and I really like its features, but I can't fully operate it as I would like because the Spanish language is not available as an option for the user interface. Could you add the Spanish language to the UI of the program?

**Comment by tnodir** [comment]
Hello.

Let me add Spanish to https://crowdin.com/project/fort-firewall

So you can translate there.

**Comment by tnodir** [comment]
https://crowdin.com/project/fort-firewall/es-ES

Please feel free to translate to Spanish.

**Reply by laramaceo** [reply]
I've gone to the place, but I'm not capable enough to do this job, I just don't know how to do it. I live in Cuba, my Internet bandwidth is very limited and because of mobile data. I need this software, could you commission the translation to someone who speaks Spanish to do this work? I would be VERY GRATEFUL AND HAPPY.

**Reply by tnodir** [reply]
> could you commission the translation to someone who speaks Spanish to do this work?

Sorry, I don't know anyone like that.

**Reply by laramaceo** [reply]
:(  I will send the translation request to Telegram friends, Linux users, maybe they can help with the translation. Thank you.

**Reply by laramaceo** [reply]
Is the translation made on the site you indicated valid for both Windows and Linux?

**Reply by tnodir** [reply]
> Is the translation made on the site you indicated valid for both Windows and Linux?

What do you mean?

1. Fort Firewall works only on Windows.
2. You can work on any OS (Windows, Linux, Mac etc) to open the site to translate.

**Reply by laramaceo** [reply]
I'm using the Windows version at the moment; is the translation file created at https://crowdin.com/project/fort-firewall/es-ES , valid for Windows OS?

**Reply by tnodir** [reply]
Sure, it's valid for Windows.

**Comment by laramaceo** [comment]
Hello.
I am already translating the Fort Firewall user interface. I am making an effort to make the translation as good as possible. I am a common user with no computer or network knowledge.
I am going to rectify the Spanish translations of the Titles. In most cases, the initial capital letters of the words in the titles, I have put them in lower case. I did not realize that I had to imitate how you had written it. This will affect the translations already approved. An example to help you understand.

Include    All

Incluir    todo 


![Rectificar inicial mayúscula de los títulos ](https://github.com/user-attachments/assets/ccb0b687-d293-4271-9e9d-c03e505e8a60)

**Reply by tnodir** [reply]
> I did not realize that I had to imitate how you had written it.

No, you have not to imitate it.

Please use appropriate casing of your language.

**Comment by laramaceo** [comment]
Is it necessary to keep this in the translated texts? %1&quot;

![2024-09-03_210045](https://github.com/user-attachments/assets/e17cedb8-dda4-41e3-b7bd-81cf7ae78119)

**Reply by tnodir** [reply]
Yes, the `%1` should be kept as is. Fort replaces the `%1` by text to show.

You can use `'` instead of `&quot;`.

**Comment by tnodir** [comment]
@laramaceo
> I may have to rectify some of the translations already approved.

Feel free to update the translations. No problem.

> it is necessary for me to have a preliminary version of the file with .qm extension, to see directly in the user interface

You can do it already by:
- click "Download" on "Spanish" language in [Crowdin](https://crowdin.com/project/fort-firewall/es-ES)
- save the `i18n_es_ES.ts` as `i18n_es.ts` file somewhere
- open the `i18n_es.ts` file in "Qt Linguist" program [1]
- click the File: "Release" menu item to create `i18n_es.qm` file
- save the `i18n_es.qm` file in the `C:\Program Files\Fort Firewall\i18n` folder
- restart "Fort Firewall" client

[1] You can download "Qt Linguist" from [thurask/Qt-Linguist](https://github.com/thurask/Qt-Linguist/releases/latest):
- unarchive the `linguist_6.7.2.zip` and run `linguist.exe`

**Comment by laramaceo** [comment]
# All programs in the sub-path:

Is it necessary to keep “#” in the text?

**Reply by tnodir** [reply]
Yes, the `#` should be at start of text.

**Comment by laramaceo** [comment]
![Screenshot 2024-09-05 5 34 02 PM](https://github.com/user-attachments/assets/5852ecd9-1a44-4811-8a1c-48a175395d7f)

**Reply by tnodir** [reply]
Yes, please keep the `#` at start of text.

**Comment by laramaceo** [comment]
I am using GPT-4o mini, and I give instructions to translate texts taking into account the content of a firweall and networks.

I use formal expressions.

I keep as much as possible the same terminology throughout the text.

**Comment by laramaceo** [comment]
As much as possible, I keep the same terminology throughout the UI text. That is what I meant.

**Comment by tnodir** [comment]
Added in [v3.13.10](https://github.com/tnodir/fort/releases/tag/v3.13.10).
